infra/gcp/ensure-staging-storage.sh

```diff
@@ -311,3 +311,12 @@ color 6 "Configuring special case for k8s-staging-releng-test"
 (
     ensure_staging_gcb_builder_service_account "releng-test" "k8s-infra-prow-build"
 )
+
+# Special case: In order to build the node images using image-builder it needs
+#               the compute api to be enabled because it will create a VM
+#               to build the node image.
+color 6 "Configuring special case for k8s-staging-cluster-api-gcp"
+(
+    readonly STAGING_PROJECT="k8s-staging-cluster-api-gcp"
+    enable_api "${STAGING_PROJECT}" compute.googleapis.com
```

Comment:
```suggestion
    readonly STAGING_PROJECT="k8s-staging-cluster-api-gcp"
    enable_api "${STAGING_PROJECT}" compute.googleapis.com
    ensure_staging_gcb_builder_service_account "cluster-api-gcp" "k8s-infra-prow-build-trusted"
```

You'll need to add a service account file for this, as you did in https://github.com/kubernetes/k8s.io/pull/1620 (just the yaml file).  Then update the job definition to use the service account, as you did in https://github.com/kubernetes/test-infra/pull/20703

Longer term, this is another special case of https://github.com/kubernetes/test-infra/issues/20652#issuecomment-774233051 (gcp-builder per staging project) cropping up again.  We should implement this across the board this release cycle IMO.

Reply:
@spiffxp Looking at https://www.packer.io/docs/builders/googlecompute, `cluster-api-gcp` will need more permissions to create the VM disks.

Reply:
ahhh!

Reply:
@spiffxp @ameukam thanks forgot that :( added the suggestions and update the job in test-infra: https://github.com/kubernetes/test-infra/pull/22041

Reply:
regarding this comment:
> Longer term, this is another special case of kubernetes/test-infra#20652 (comment) (gcp-builder per staging project) cropping up again. We should implement this across the board this release cycle IMO.

I will work on that for this cycle but will need some initial help